### PR TITLE
Change the order verification

### PR DESCRIPTION
**Pull request by GNUp**
Make order verification feasible when `asset_type_to` or `asset_type_from` is same with `asset_type_fee`.

### CODE REVIEW

**Comment by GNUp** [comment]
The integration tests in the order part will fail until the SDK is changed.

File: types/src/transaction/order.rs [hunk 1]
```diff
@@ -83,6 +90,7 @@ impl Order {
                 fee: self.asset_quantity_fee,
             })
         }
+        // fee recipient should not be same with the one provided asset_type_from
```
Comment (majecty) [inline types/src/transaction/order.rs]: `be same with` -> `be the same with`

**Comment by majecty** [comment]
I can't find any semantic problems in the commits.
However, it will take quite a while to verify that this code checks all the verification needed in the order. Since the order change will be released in version 2.0, let's merge this PR and check the verifications all again before the release of the 2.0.

File: types/src/transaction/action.rs [hunk 2]
```diff
@@ -191,7 +191,7 @@ impl Action {
                 ..
             } => {
                 if output.supply == 0 {
-                    return Err(SyntaxError::ZeroQuantity)
+                    return Err(SyntaxError::ZeroQuantity);
```
[the remark this answers is not in the document]
Comment (GNUp) [reply types/src/transaction/action.rs]: My vscode autoformatting changed it. I fix it.

**Comment by GNUp** [comment]
Okay, then let's merge this after fixing the integration test.

**Comment by sgkim126** [comment]
@majecty Please keep this commit unmerged after review until we finish to design the service that uses Order.

**Comment by GNUp** [comment]
The integration test of this PR is using now using [this](https://github.com/GNUp/codechain-sdk-js/tree/orderBuild), which can be merged into the master branch of the SDK after merging this PR.

**Comment by sgkim126** [comment]
@majecty Let's merge it and test it on the testnet.

**Comment by sgkim126** [comment]
@GNUp Please remove the WIP tag.

**Comment by majecty** [comment]
@GNUp ping

**Pull request merged**